internal/services/redis/redis_cache_resource.go

```diff
@@ -179,6 +179,16 @@ func resourceRedisCache() *pluginsdk.Resource {
 							Computed: true,
 						},
 
+						"preferred_data_persistence_auth_method": {
```

Comment:
Abbreviations should be spelled out and `preferred` appears superfluous here, could we please rename this to
```suggestion
						"data_persistence_authentication_method": {
```

Reply:
Hi @stephybun,

I just tried to be consistent with the API responses. Below is what I have from API ::

```
"redisConfiguration": {
      "maxclients": "1000",
      "maxfragmentationmemory-reserved": "125",
      "maxmemory-delta": "125",
      "maxmemory-policy": "volatile-lru",
      "maxmemory-reserved": "125",
      "preferred-data-persistence-auth-method": "ManagedIdentity"
},
```

If you still want it to be renamed to `data_persistence_authentication_method`, I would do it.

Reply:
Since we don't tend to use abbreviations I think it would be good if we renamed this `data_persistence_authentication_method` , thanks @harshavmb!

Reply:
Done @catriona-m